common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/ExternalBlockHandler.java

```diff
@@ -323,10 +323,13 @@ private void checkAuth(TransportClient client, String appId) {
 
     public ShuffleMetrics() {
       allMetrics = new HashMap<>();
-      allMetrics.put("openBlockRequestLatencyMillis", openBlockRequestLatencyMillis);
-      allMetrics.put("registerExecutorRequestLatencyMillis", registerExecutorRequestLatencyMillis);
-      allMetrics.put("fetchMergedBlocksMetaLatencyMillis", fetchMergedBlocksMetaLatencyMillis);
-      allMetrics.put("finalizeShuffleMergeLatencyMillis", finalizeShuffleMergeLatencyMillis);
+      // Note that for the latency metrics, the default unit is actually nanos, not millis.
+      // The variables have been renamed, but to preserve backwards compatibility, the metric
+      // names remain unchanged. See SPARK-35259 for more details.
+      allMetrics.put("openBlockRequestLatencyMillis", openBlockRequestLatency);
```

Comment:
Ur, this is not what I asked. I must be clearer. I asked to expose the correct metric value which mean converting the value to `mills` and put here, @xkrogen .
> -1 for removing the existing metrics like openBlockRequestLatencyMillis. Instead, we need to expose correct metric values and a migration doc.

Reply:
@dongjoon-hyun It requires a `Metric` here rather than a long value.

The `Timer` doesn't seem to provide any APIs to get the milliseconds. The only way I see now is to implements Spark's own `Timer` by extending Dropwizard's one.

Reply:
Ohh, I understand now @dongjoon-hyun. Thanks for the clarification.

For the YARN case specifically, we can modify `YarnShuffleServiceMetrics` to do the conversion. This would be quite straightforward.

However I'm not sure of how the metrics are used for the non-YARN case, so I'm not sure what would be appropriate. Can either of you comment on that?

Reply:
@Ngone51 . Don't get me wrong. What I suggested was a simple conversion to keep the existing metric names while fixing the wrong metric values. Do we need more from Spark-side?
```
allMetrics.put("openBlockRequestLatencyMillis", openBlockRequestLatency / 1000);
```

Reply:
Did I miss something at the original issue? I'm confused due to the complexity of this PR's proposal and @Ngone51 's suggestion.

Reply:
I thought on this a bit more and came to agree with @Ngone51 that the better way to do this is to implement a custom `Timer` instead of trying to perform the conversion at some other layer. Just put up a new diff for that.